book/src/dyn_tutorial/permissions.md

```diff
@@ -0,0 +1,126 @@
+| ------------ | ---------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `my`         | Unique ownership                   | Nobody else has access to this object unless we grant it to them.                                                                                                                                                            |
+| `our`        | [Shared ownership](./share.md)     | Many variables can share access at once, but [none of them can write to the object](./sharing_xor_mutation.md).                                                                                                              |
+| `leased`     | [Unique lease](./lease.md)         | the object is [leased](./lease.md) from another variable. This variable has unique access, but it is only temporary. The owner can cancel the lease and start using the object again.                                        |
+| `our leased` | [Shared lease](./shared_leases.md) | a [shared lease](./shared_leases.md) means that multiple variables have ([read-only](./sharing_xor_mutation.md)) access, but that access is only temporary. The owner can cancel the lease and start using the object again. |
+| `any`        |                                    | the variable can store values with any of the above permisions                                                                                                                                                               |
+
+## Shared ownership
+
+For example, instead of writing `my p`, we could've written `our p` to declare a variable with shared ownership:
+
+```
+class Point(x, y)
+
+async fn main() {
+    our p = Point(22, 44)
+}
+```
+
+
+## Enforcing uniqueness and ownership
+
+When you declare a variable as `my p`, that means that it has *unique ownership* of the object(s) stored inside of it. 
+
+```
+async fn main() {
+    our p = Point(22, 44)
+    my q = p                                         # Error
+}
+```
+
+
+
+## Permissions and mutation
+
+The most obvious effect 
+## Default permissions
+
+If you don't specify any ownership permision, the default is `our leased`. This means that the variable cannot be used to mutate the fields of the object. Example:
+
+```
+class Point(x, y)
+
+async fn main() {
+    p = Point(22, 44)      # declares `p` in `our leased` mode
+    p.x += 1               # Error!
+}
+```
+
+## Ownership permissions are enforced
+
+We saw that `my p` indicates that the variable `p` has unique ownership of its contents. But what happens if we try to store a value in `p` that cannot be owned? As an example, integers like `22` and `44` are inherently shared. If we try to claim ownership of those, we get an error. As an example, try running this:
```

Comment:
The idea that integers are inherently shared is not obvious to me. You've explained it me previously but I can't recollect the reasoning. It might be good to expand on that.

Intuition tells me I should be able to own and mutate any value.

Reply:
One additional consideration here for me - I know the sharing mechanism hasn't been explained in the text, but considering that `our` types are meant to be reference counted, this suggests to be that integers are reference counted. I doubt that is the case, but I'm curious what distinguishes integers from other types to avoid putting them on the heap.

Reply:
I don't intend to ref-count every integer, no, but I *do* intend that you can *think of things* as having every integer be a ref-counted object (much like it would be in Python).

Reply:
> Intuition tells me I should be able to own and mutate any value.

Just to be clear, you can own the value, you just can't mutate an integer's fields (it doesn't have any). It's like you own an `Rc<i32>`, essentially. 

Maybe it helps for me to say that I expected to have `our class Point(our x, our y)` as a way to declare your own class that is "always shared". In this case, you could never have a `my p = Point(22, 44)`, because nobody can have single ownership of a `Point`; you could have `our p = Point(22, 44)`, and you could also do `p := Point(44, 66)` to change that to another point.

Reply:
Thanks. I think I understand. I need to stop trying to map Rust's `let` and `let mut` onto Dada's permissions.